fairlearn/datasets/_synthetic.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) Fairlearn contributors.
+# Licensed under the MIT License.
+import numpy as np
+from sklearn.datasets import make_classification
+from sklearn.utils import check_random_state
+
+
+def make_synthetic_dataset(classes=None, n_features=20, n_informative=4, random_state=None):
+    """Create a synthetic dataset with a single sensitive feature: 'gender'.
+
+    Parameters
+    ----------
+    classes : dict, default=None
+        A dict whose keys are the labels for the gender feature, and whose set
+        the `n_samples` and `class_sep` parameters for `make_classification`.
+        If `None`, the following default dict is used::
```

Comment:
`defaultdict` is actually a utility that's part of `collections` (and quite useful!), so perhaps I'd rephrase to
```
If `None`, the following dictionary is used by default:
```
The double colon is probably there by accident, right? (or is it special syntax for ReST?)

Reply:
I've reworded as you suggested. The double colon is the ReST syntax for making the next paragraph a literal block.

Reply:
BTW, I don't really like that I've duplicated this code into the docstring, but I'm hoping things can be sufficiently cleaned up in the next PR so that this will become unnecessary.

Reply:
Oh neat, I didn't know about that one! 🙂 

I agree 100% on the duplication (and the future removal), but having good small examples in the API docs is also a priority.